halo2_proofs/src/circuit/value.rs

```diff
@@ -14,7 +14,7 @@ use crate::plonk::{Assigned, Error};
 ///   for improved usability.
 #[derive(Clone, Copy, Debug)]
 pub struct Value<V> {
-    inner: Option<V>,
+    pub inner: Option<V>,
```

Comment:
This has been discussed before and we prefer to do it by stealing from `map(…)`

[Code in zkevm-circuits](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/58995d3eca65fa2056713bdda04bf70910393974/zkevm-circuits/src/keccak_circuit/util.rs#L236)

```rust
pub(crate) fn extract_field<F: FieldExt>(value: Value<F>) -> F {
    let mut field = F::zero();
    let _ = value.map(|f| {
        field = f;
        f
    });
    field
}
```